Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -188,19 +186,17 @@ public sealed class VsProjectAnalyzer : IDisposable {
             entry.Properties[_pathInZipFile] = value;
         }
 
-        private void OnModulesChanged(object sender, EventArgs e) {
+        private async void OnModulesChanged(object sender, EventArgs e) {
             Debug.Assert(_pyAnalyzer != null, "Should not have null _pyAnalyzer here");
             if (_pyAnalyzer == null) {
                 return;
             }
 
-            lock (_contentsLock) {
-                _pyAnalyzer.ReloadModulesAsync().WaitAndUnwrapExceptions();
+            await _pyAnalyzer.ReloadModulesAsync();

```

Comment:
Given that this is async now, it seems that it'll have to deal with the possibility of _pyAnalyzer being disposed concurrently.